File: stepik-java-api/src/main/java/org/stepik/api/queries/stepiks/StepikStepiksGetQuery.java
package org.stepik.api.queries.stepiks;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.stepiks.Stepiks;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikStepiksGetQuery extends StepikAbstractGetQuery<StepikStepiksGetQuery, Stepiks> {
    private int id;

    public StepikStepiksGetQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Stepiks.class);
    }

    @NotNull
    public StepikStepiksGetQuery id(int value) {
        id = value;
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.STEPIKS + "/" + id;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/stepiks/StepikStepiksGetQuery.kt
package org.stepik.api.queries.stepiks

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.stepiks.Stepiks
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikStepiksGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikStepiksGetQuery, Stepiks>(stepikAction, Stepiks::class.java) {
    
    private var id: Int = 0
    
    fun id(value: Int): StepikStepiksGetQuery {
        id = value
        return this
    }
    
    override fun isCacheEnabled() = false
    
    override val url = "${stepikAction.stepikApiClient.host}/api/stepics/$id"
    
}
